### PR TITLE
AC_AttitudeControl: PosControl: split the accel limits into up and down components

### PR DESCRIPTION
**Pull request by Charlie-Burge**
Split the accel limits to separately to control the up limit and down limit

### CODE REVIEW

**Review by Dan-Moss — APPROVED** [review]

File: libraries/AC_AttitudeControl/AC_PosControl.cpp [hunk 1]
```diff
@@ -312,14 +312,23 @@ const AP_Param::GroupInfo AC_PosControl::var_info[] = {
...
+    // @DisplayName: Maximin vertical climb acceleration
+    // @Description: Hard vertical climb acceleration limit, must be larger than waypoint and pilot targets, 0 disables
     // @Units: m/s/s
     // @Range: 0.5 5
     // @Increment: 0.1
     // @User: Advanced
-    AP_GROUPINFO("_ACCEL_Z_MAX", 14, AC_PosControl, _accel_z_max, 0),
+    AP_GROUPINFO("_ACCEL_UP_MAX", 14, AC_PosControl, _accel_z_up_max, 0),
+
+    // @Param: _ACCEL_Z_DN_MAX
```
Comment (IamPete1) [inline libraries/AC_AttitudeControl/AC_PosControl.cpp]:
```suggestion
    // @Param: _ACCEL_DN_MAX
```

File: libraries/AC_AttitudeControl/AC_PosControl.cpp [hunk 2]
```diff
@@ -312,14 +312,23 @@ const AP_Param::GroupInfo AC_PosControl::var_info[] = {
     // @User: Advanced
     AP_GROUPINFO("_SPEED_DN_MAX", 13, AC_PosControl, _speed_down_max, 0),
 
-    // @Param: _ACCEL_Z_MAX
-    // @DisplayName: Maximin vertical acceleration
-    // @Description: Hard vertical acceleration limit, must be larger than waypoint and pilot targets, 0 disables
+    // @Param: _ACCEL_Z_UP_MAX
```
Comment (IamPete1) [inline libraries/AC_AttitudeControl/AC_PosControl.cpp]:
```suggestion
    // @Param: _ACCEL_UP_MAX
```

File: ArduCopter/AP_Arming.cpp [hunk 3]
```diff
@@ -170,23 +170,31 @@ bool AP_Arming_Copter::parameter_checks(bool display_failure)
...
-            check_failed(ARMING_CHECK_PARAMETERS, display_failure, "WPNAV_SPEED_UP too close to PSC_SPEED_UP_MAX");
-            return false;
+           check_failed(ARMING_CHECK_PARAMETERS, display_failure, "WPNAV_SPEED_UP too close to PSC_SPEED_UP_MAX");
+           return false;
         }
         if (!copter.pos_control->target_speed_down_within_limit(copter.wp_nav->get_default_speed_down()*0.01)) {
-            check_failed(ARMING_CHECK_PARAMETERS, display_failure, "WPNAV_SPEED_DN too close to PSC_SPEED_DN_MAX");
-            return false;
+           check_failed(ARMING_CHECK_PARAMETERS, display_failure, "WPNAV_SPEED_DN too close to PSC_SPEED_DN_MAX");
+           return false;
```
Comment (IamPete1) [inline ArduCopter/AP_Arming.cpp]: These were indented correctly I think.

**Comment by IamPete1** [comment]
I don't really understand the point of separate accel limits with a single accel target, but this change basically looks good.

**Pull request merged**